sgt-back-end/index.js

```diff
@@ -0,0 +1,149 @@
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = req.body.score;
+  if (!name || !score || !course) {
+    res.status(400).json({
+      error: 'name, course, and score are required'
+    });
+    return;
+  } else if (parseInt(score) !== Number(score) || score > 100 || score < 1) {
+    res.status(400).json({
+      error: 'score must be a an integer in range of 1 and 100'
+    });
+    return;
+  }
+  const sql = `
+              INSERT INTO grades (name, course, score)
+              VALUES ($1, $2, $3)
+           RETURNING *
+            `;
+  const values = [name, course, score];
+  db.query(sql, values)
+    .then(result => {
+      const newStudent = result.rows[0];
+      res.status(201).json(newStudent);
+    })
+    .catch(err => {
+      console.error(new Error(err));
+      res.status(500).json({
+        error: 'an unexpected error occured'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = req.body.score;
+  const gradeId = req.params.gradeId;
+  if (!name || !course || !score) {
+    res.status(400).json({
+      error: 'name, course, and score are required'
+    });
+    return;
+  } else if (parseInt(score) !== Number(score) || score > 100 || score < 1) {
+    res.status(400).json({
+      error: 'score must be an integer in the range 1 to 100'
+    });
+    return;
+  } else if (parseInt(gradeId) !== Number(gradeId) || gradeId < 1) {
+    res.status(400).json({
+      error: 'gradeId must be an integer greater than 0'
+    });
+  }
+  const sql = `
+    UPDATE grades
+       SET name = $1,
+           course = $2,
+           score = $3
+     WHERE "gradeId" = $4
+ RETURNING *
+  `;
+  const values = [name, course, score, gradeId];
+  db.query(sql, values)
+    .then(result => {
+      const updatedStudent = result.rows[0];
+      if (!updatedStudent) {
+        res.status(404).json({
+          error: 'gradeId not found'
+        });
+      } else {
+        res.status(200).json(updatedStudent);
+      }
+    })
+    .catch(err => {
+      console.error(new Error(err));
+      res.status(500).json({
+        error: 'an unexpected error occured'
+      });
+    });
+});
+
+app.delete('/api/grades/:gradeId', (req, res) => {
+  const gradeId = req.params.gradeId;
+  if (!gradeId || gradeId < 1 || parseInt(gradeId) !== Number(gradeId)) {
+    res.status(400).json({
+      error: 'gradeId is required and must be a positive integer'
+    });
+  }
+  const sql = `
+    DELETE FROM grades
+          WHERE "gradeId" = $1
+      RETURNING *
+  `;
+  const values = [gradeId];
+  db.query(sql, values)
+    .then(result => {
+      const deletedStudent = result.rows[0];
+      if (!gradeId) {
+        res.status(404).json({
+          error: 'gradeId not found'
+        });
+      }
+      res.status(204).json(deletedStudent);
```

Comment:
When you use status `204` that means "No Content", so sending data won't work, the client will receive an empty body. So when using status `204` it is better to use `sendStatus`.
```JAVASCRIPT
res.sendStatus(204);
```